prov/monitoring/check.go

```diff
@@ -110,6 +116,15 @@ func (c *Check) disable() {
 	}
 }
 
+func (c *Check) enable() {
+	if !c.enabled {
```

Comment:
I'm surprised to not see this under lock sync
Is it safe for concurrent use?

Reply:
Hum, I thougt that there was no need to handle any concurrency for this one but looking closer, concurrency can happen when stopping the check